src/components/HomePage/SearchResultsArea/Offer/OfferInteractionOptions.js

```diff
@@ -45,7 +48,18 @@ const OfferVisibilityOptions = ({
         }
     };
 
-    if (!loading && (sessionData?.isAdmin || sessionData?.company?._id === offer.owner))
+    const handleApplyURLRedirect = () => {
+        setClosedRedirectDialog();
+        window.open(offer.applyURL, "_blank", "noopener");
+    };
```

Comment:
You should use useCallback otherwise the component will always be redefining that function each render, causing the lower component to rerender aswell.

https://reactjs.org/docs/hooks-reference.html#usecallback